tests/test_llama.py

```diff
@@ -224,6 +225,25 @@ def test_llama_lm_head_model(num_kv_heads):
     assert out.array.shape == (Batch.size, Pos.size, Vocab.size)
 
 
+@pytest.mark.parametrize("use_flash", [True, False])
+@pytest.mark.parametrize("num_kv_heads", [1, 2, 4])
+def test_llama_lm_head_model_bwd(use_flash, num_kv_heads):
+    llama_config = _get_llama_config(use_flash=use_flash, num_kv_heads=num_kv_heads)
+    Batch = hax.Axis("batch", 2)
+    Vocab = hax.Axis("vocab", 1000)
+    Pos = llama_config.Pos
+    input_ids = hax.random.randint(random.PRNGKey(0), (Batch, Pos), 0, Vocab.size)
+    mask = hax.nn.attention.causal_mask(Pos, llama_config.KeyPos)
+
+    llama_model = LlamaLMHeadModel.init(Vocab=Vocab, config=llama_config, key=random.PRNGKey(0))
+
+    def f(llama_model, input_ids, mask):
+        out = llama_model(input_ids, mask)
+        return hax.sum(out).scalar()
+
+    _, grads = eqx.filter_value_and_grad(f)(llama_model, input_ids, mask)
```

Comment:
can we add in test_fa that checks for all_close in the FA-vs-non-FA case for GQA? I suspect this code is incorrect

Reply:
sorry I did made a mistake there.
Added a grad test in test_fa and now it passes.